src/raw_iter.rs

```diff
@@ -112,50 +151,72 @@ enum RleState {
     EndOfBitmap,
 }
 
+pub struct PixelPoints {
+    /// The location of the next pixel
+    next_pixel: Point,
+    /// The width of a line in pixels
+    width: u32,
+}
+
 /// Iterator over individual BMP RLE8 encoded pixels.
 ///
 /// Each pixel is returned as a `u32` regardless of the bit depth of the source image.
 #[derive(Debug)]
-pub struct Rle8Pixels<'a> {
+pub struct Rle8Colors<'a> {
     /// Our source data
     data: &'a [u8],
     /// Our state
     rle_state: RleState,
-    /// The width of a line in pixels
-    width: u32,
-    /// The location of the next pixel
-    next_pixel: Point,
+    start_of_line: bool,
 }
 
-impl<'a> Rle8Pixels<'a> {
-    /// Create a new RLE pixel iterator.
-    pub(crate) fn new(raw_bmp: &RawBmp<'a>) -> Rle8Pixels<'a> {
-        let header = raw_bmp.header();
-        Rle8Pixels {
-            data: raw_bmp.image_data(),
-            rle_state: RleState::Starting,
-            width: header.image_size.width,
-            // RLE encoded bitmaps are upside down
-            next_pixel: Point::new(0, (header.image_size.height - 1) as i32),
+impl PixelPoints {
+    fn next_up_dir(&mut self) -> Point {
```

Comment:
Instead of having two separate `next` methods for bottom up and top down I would suggest to add a `row_order` or `delta_y` field to the struct. This way `PixelPoints` can be a real `Iterator` and we can get rid of the `match` in `RawPixels::next` to get a minimal performance benefit by only having to check the row order at the end of each line and not for each pixel.